src/Agent.Worker/Build/BuildDirectoryManager.cs

```diff
@@ -81,19 +81,24 @@ public sealed class BuildDirectoryManager : AgentService, IBuildDirectoryManager
 
         public TrackingConfig PrepareDirectory(
             IExecutionContext executionContext,
-            RepositoryResource repository,
+            IList<RepositoryResource> repositories,
             WorkspaceOptions workspace)
         {
             // Validate parameters.
             Trace.Entering();
             ArgUtil.NotNull(executionContext, nameof(executionContext));
             ArgUtil.NotNull(executionContext.Variables, nameof(executionContext.Variables));
-            ArgUtil.NotNull(repository, nameof(repository));
+            ArgUtil.NotNull(repositories, nameof(repositories));
+
+            var selfRepository = repositories.FirstOrDefault(r => string.Equals(r.Alias, "self", StringComparison.OrdinalIgnoreCase));
+            ArgUtil.NotNull(selfRepository, nameof(selfRepository));
```

Comment:
What happens when this runs for a designer build?

Reply:
Great question! Found that problem today because of your review of my server code. I have fixed the code up so that shouldn't be a problem. The solution is basically to always take the first repo if there is only one.